### PR TITLE
Train to be a teacher

### PR DESCRIPTION
**Pull request by peteryates**
### Trello card

https://trello.com/c/H4N1jvTT/3130-build-new-navigation-pages

### Context

This is the first proper attempt at adding a second level of navigation to the site. The previous attempts relied on drop-down menus that were a bit of a pain from a usability perspective, this one has 'category pages' that contain a card per category.

The files should be structured in the following manner:

```
app/views/content/
  - train-to-be-a-tacher.md     (the category page)
  + train-to-be-a-tacher        (a directory with matching name)
    - my-first-page.md          (pages within the cateogry)
    - my-second-page.md
    - _my-cta.html.erb          (partials placed on the category page)
```

Prior work was done in #2401, the content is discussed there at great length.

### Changes

- Add `Pages::Navigation#find` to select nodes
- Add category page with cards for each section
- Remove menu functionality from nav component

### Remaining tasks

* [x] Move the content over from #2401 once it's finalised
* [x] Update the hero so it contains the blurb beneath the title
* [x] Add the 'Questions' section and links
* [x] Remove 'Ways to train' from top nav
* [x] Remove 'Ways to train' from footer links
* [x] Update the pink style of the 'also on this page' block
* [x] Help and advice page
* [ ] Set up any redirects needed (eg Ways to train?, SKE?, Get school experience?)
* [x] Lighthouse tests, usability review
* [x] Make the events promo search actually work
* [x] Replace 'Chat online' link with component
* [x] Replace the £ icon with a 🔍 (see [here](https://github.com/DFE-Digital/get-into-teaching-app/blob/new-nav/app/views/content/train-to-be-a-teacher/promos/_find-your-undergraduate-course.html.erb) and possibly [here](https://github.com/DFE-Digital/get-into-teaching-app/blob/new-nav/app/views/content/train-to-be-a-teacher/promos/_find-your-course.html.erb)) 

### Preview

| Desktop | Mobile |
| ----------- | --------|
| Screenshot from 2022-04-20 16-28-51 | Screenshot from 2022-04-20 16-29-14 |

### CODE REVIEW

**Comment by peteryates** [comment]
Note - I've left the hero background grey in this branch. It looks/feels weird when one is a different colour to the others, plus it's another variation to maintain.

File: app/views/content/train-to-be-a-teacher/if-you-have-a-degree.md
Comment (EllieNodder) [inline app/views/content/train-to-be-a-teacher/if-you-have-a-degree.md:52]: @valentine-carter we're a bit inconsistent with capitalising School Direct - should it be School direct or School Direct?

Comment (valentine-carter) [reply app/views/content/train-to-be-a-teacher/if-you-have-a-degree.md:52]: Typo, innit. School Direct.

File: app/views/layouts/content.html.erb
Comment (gemmadallmandfe) [inline app/views/layouts/content.html.erb:69]:
```suggestion
              <h2 class="link-block__header">Related content</h2>
```

File: app/views/content/help-and-advice.md
Comment (gemmadallmandfe) [inline app/views/content/help-and-advice.md:67]: @peteryates Does this chat button on the Help and advice page have the usual chat functionality that it is only surfaced during hours when chat is available, otherwise there is a closed message?

Plus does it have the email fallback for non-JS users?

Comment (peteryates) [reply app/views/content/help-and-advice.md:67]: Didn't even realise we had a chat online button here. It has none of that currently (just a link) - I think we can drop the ChatComponent in here which will have all that functionality. Will check.

**Comment by gemmadallmandfe** [comment]
@EllieNodder I'm sure you are aware of this already but a lot of the pages are missing titles/meta descriptions?

**Comment by EllieNodder** [comment]
> @EllieNodder I'm sure you are aware of this already but a lot of the pages are missing titles/meta descriptions?

Yep, we were holding off on those in case any of the pages/content changed. Shall be doing those today.

File: app/views/content/train-to-be-a-teacher/_questions.html.erb
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/_questions.html.erb:27]: @peteryates I think it's the actual page URL that needs amending, not the path on this button - we need to keep with the URLs that we confirmed to the marketing team (see https://trello.com/c/MQgtSwLn)

So this initial teacher training page needs to have a path of /train-to-be-a-teacher/initial-teacher-training ?

Comment (peteryates) [reply app/views/content/train-to-be-a-teacher/_questions.html.erb:27]: Ah! Didn't realise, will amend

File: app/views/content/train-to-be-a-teacher/promos/_find-your-course.html.erb
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/promos/_find-your-course.html.erb:4]: @peteryates Is a black magnifying glass icon something that @gazcarr can help with ?

File: app/views/content/train-to-be-a-teacher/_questions.html.erb
Comment (peteryates) [reply app/views/content/train-to-be-a-teacher/_questions.html.erb]: > So this initial teacher training page needs to have a path of /train-to-be-a-teacher/initial-teacher-training

@gemmadallmandfe - this is where I added the ITT card 🙈

**Comment by peteryates** [comment]
The new 'Why teach' section:

| Desktop | Mobile | 
| ------- | --------|
| Screenshot from 2022-04-27 12-45-31 | Screenshot from 2022-04-27 12-45-23 |

**Comment by gemmadallmandfe** [comment]
@peteryates Myles has just flagged to me that this is some additional whitespace around the mailing list bar (you might already be aware of this!)

image

**Comment by peteryates** [comment]
@gemmadallmandfe - not sure why but the green bar has a `3em` margin explicitly set on the bottom.

Screenshot from 2022-04-27 13-53-37

Perhaps we should ask the author why it's there 😂 cc @MylesJarvis

**Comment by peteryates** [comment]
Green bar fixed in #2479 😄

File: app/views/content/help-and-advice.md
Comment (peteryates) [inline app/views/content/help-and-advice.md:8]:
```suggestion
title_paragraph: We’re here to answer your questions and provide advice about teacher training, whether you’re just thinking about teaching or you’re ready to apply.
```

File: app/views/content/train-to-be-a-teacher/_why-teach.html.erb
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/_why-teach.html.erb:16]: @valentine-carter Is this link text a bit misleading until we have more Why teach content? It just goes to the salaries page which doesn't seem to quite fit with the Inspire the next generation message of this section?

Comment (valentine-carter) [reply app/views/content/train-to-be-a-teacher/_why-teach.html.erb:16]: Indeed. I need to change the copy on this, it's still the mock copy. I just need to fix my ports not loading first.

File: app/views/content/train-to-be-a-teacher/_why-teach.html.erb
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/_why-teach.html.erb:16]:
```suggestion
        <li><%= link_to("Find out what pay and benefits teachers get", "/salaries-and-benefits") %></li>
```

File: app/views/content/help-and-advice.md
Comment (gemmadallmandfe) [inline app/views/content/help-and-advice.md:56]: @gazcarr @EllieNodder 

Feedback from Steph in Candidate Experience team - should we make the phone number more prominent (as it is in the Talk to us footer section)?

On mobile it is a link anyway - just whether you think it is worth doing anything that caters for desktop traffic (your call)

Comment (EllieNodder) [reply app/views/content/help-and-advice.md:56]: image
@gazcarr @gemmadallmandfe we could put the phone number on a new line in bold like this?

File: app/views/content/help-and-advice.md
Comment (gemmadallmandfe) [inline app/views/content/help-and-advice.md:34]:
```suggestion
You could also attend a school and university event, either online or in person, and hear directly from teacher training providers about the courses they offer and how to apply.
```

@EllieNodder Suggested change from the Events team for you to consider

NB once we have the new events design live, these events will be called 'Training provider events' rather than 'School and university events'

File: app/views/content/train-to-be-a-teacher/get-school-experience.md
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/get-school-experience.md:9]:
```suggestion
    Train to Teach events : "/event-categories/train-to-teach-events"
```

@EllieNodder Just flagging this - we need to capitalise 'Teach' wherever referenced as it is the brand name of their events

File: app/views/content/train-to-be-a-teacher/if-you-have-a-degree.md
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/if-you-have-a-degree.md:59]: @EllieNodder I don't think this link text is quite right - it takes you to Find

ie it is more about Find courses including SD salaried?

File: app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb:4]: @EllieNodder Is it too long a title to have 'National Virtual Train to Teach event'?

Trying to emphasise it is for everyone.....

File: app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb:7]: @EllieNodder Think we should probably try and highlight that it is not just a one-way presentation?

Eg something like 'Hear from training providers, chat with teachers and get one-to-one advice from expert advisers.'

(but have a look at the event details and feel free to change it!)

File: app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb
Comment (EllieNodder) [reply app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb:4]: @gemmadallmandfe felt like national virtual was a bit of a mouthful, and the fact that it's virtual should already emphasise it is for everyone (but equally happy to add it in)

File: app/views/content/help-and-advice.md
Comment (gemmadallmandfe) [reply app/views/content/help-and-advice.md:56]: @EllieNodder That looks good to me but I will defer to @gazcarr !

File: app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb
Comment (gemmadallmandfe) [reply app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb:4]: @EllieNodder Fair shout - let's leave as is

File: app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/promos/_events-near-you.html.erb:7]: @EllieNodder Presume this is needed!

```suggestion
    <p>Hear from training providers, chat with teachers and get one-to-one advice from expert advisers.<p>
```

**Review by EllieNodder — APPROVED** [review]
Looks good to me!

**Review by gemmadallmandfe — APPROVED** [review]
Looks good to me - happy for this to be merged

**Pull request merged**
